openedx/core/djangoapps/schedules/signals.py

```diff
@@ -87,3 +64,60 @@ def create_schedule(sender, **kwargs):
 
     log.debug('Schedules: created a new schedule starting at %s with an upgrade deadline of %s',
               content_availability_date, upgrade_deadline)
+
+
+@receiver(COURSE_START_DATE_CHANGED, dispatch_uid="update_schedules_on_course_start_changed")
+def update_schedules_on_course_start_changed(sender, updated_course_overview, previous_start_date, **kwargs):
+    """
+    Updates all course schedules if course hasn't started yet.
+    """
+    if previous_start_date > timezone.now():
```

Comment:
Should this be:

```python
current_time = timezone.now()
if previous_start_date > current_time and updated_course_overview.start > current_time:
    # update schedules
```
I doubt course teams ever do this, but it's probably worth putting a guard, just in case.

Reply:
If the new start date is less than today, wouldn't we still want to update everyone's schedules?

Reply:
Addressed in https://github.com/edx/edx-platform/pull/15998.